geonode_mapstore_client/client/js/plugins/ResourceDetails/components/DetailsPermissions.jsx

```diff
@@ -0,0 +1,151 @@
+            });
+        },
+        responseToEntries: ({ response, entries }) => {
+            return response?.users.map((user) => {
+                const { permissions } =
+          entries.find((entry) => entry.id === user.pk) || {};
+                return {
+                    ...resourceToPermissionEntry("user", user),
+                    permissions
+                };
+            });
+        }
+    },
+    {
+        id: "group",
+        labelId: "gnviewer.groups",
+        request: ({ entries, groups, ...params }) => {
+            const excludeEntries = entries
+                .filter(({ type }) => type === "group")
+                .map(({ id }) => id);
+            const excludeGroups = groups.map(({ id }) => id);
+            const exclude = [...(excludeEntries || []), ...(excludeGroups || [])];
+            return getGroups({
+                ...params,
+                "filter{-group.pk.in}": exclude
+            });
+        },
+        responseToEntries: ({ response, entries }) => {
+            return response?.groups.map((group) => {
+                const { permissions } =
+          entries.find((entry) => entry.id === group.group.pk) || {};
+                return {
+                    ...resourceToPermissionEntry("group", group),
+                    permissions
+                };
+            });
+        }
+    }
+];
+
+
+const Permissions = ({
+    resourceType,
+    permissionsLoading,
+    compactPermissions,
+    onChangePermissions
+}) => {
+    const enableGeoLimits = resourceType === ResourceTypes.DATASET;
+    const isMounted = useIsMounted();
+    const [permissionsObject, setPermissionsObject] = useState({});
+
+    useEffect(() => {
+        getResourceTypes().then((data) => {
+            const resourceIndex = findIndex(data, { name: resourceType });
+            let responseOptions;
+            if (resourceIndex !== -1) {
+                responseOptions = getResourcePermissions(
+                    data[resourceIndex].allowed_perms.compact
+                );
+            } else {
+                // set a default permission object
+                responseOptions = getResourcePermissions(data[0].allowed_perms.compact);
+            }
+            isMounted(() => setPermissionsObject(responseOptions));
+        });
+    }, [availableResourceTypes]);
+    return (
+        <PermissionsComponent
+            editing
+            compactPermissions={permissionsCompactToLists(compactPermissions)}
+            entriesTabs={entriesTabs}
+            onChange={(value) =>
+                onChangePermissions(permissionsListsToCompact(value))
+            }
+            showGroupsPermissions
+            tools={
+                enableGeoLimits
+                    ? [{ Component: GeoLimits, name: "GeoLimits" }]
+                    : []
+            }
+            loading={permissionsLoading}
+            permissionOptions={permissionsObject}
+        />
+    );
+};
+
+export default connect(
```

Comment:
If this component is connected it should be in containers and not components

Reply:
Yes, but since tabComponents are used this way and they are logically grouped in components of ResourceDetails, I placed them in components folder

Reply:
ok , but other components are not connected and they are connected in the tabComponents files.